prov/efa/src/efa_domain.c

```diff
@@ -487,6 +487,69 @@ static int efa_mr_cache_init(struct efa_domain *domain, struct fi_info *info)
 	return 0;
 }
 
+/*
+ * efa_check_hmem_support determines whether an HMEM device is initialized, and
+ * whether the provider may support p2p transfers for that device. This state is
+ * used later when determining how to initiate an HMEM transfer.
+ *
+ * @param domain efa_domain to run the check/store state
+ * @return 0 on success, negative value on an unexpected error
+ */
+static int efa_check_hmem_support(struct efa_domain *domain)
```

Comment:
the function name is a little confusing, check_xxx usually return 0 or 1. Maybe something like `efa_set_domain_hmem_info`?